plugins/pulp_ostree/plugins/importers/steps.py

```diff
@@ -67,6 +70,38 @@ def storage_dir(self):
         with SharedStorage(constants.STORAGE_PROVIDER, storage_id) as storage:
             return storage.content_dir
 
+    @property
+    def repair(self):
+        return self.config.get(constants.IMPORTER_CONFIG_REPAIR, False)
+
+
+class Repair(PluginStep):
+
+    def __init__(self):
+        super(Repair, self).__init__(step_type=constants.IMPORT_STEP_REPAIR_REPOSITORY)
+        self.description = _('Repair Local Repository')
+
+    def process_main(self, item=None):
+        """
+        Repair corrupted local repository.
+        The only option to repair a corrupted repository is to re-create it.
+        Separate step because:
+          - shutil.rmtree() is very slow and should be reflected in
+            progress reporting.
+          - anticipating better tools provided by libostree for
+            doing the repair in the future.
+
+        :raises PulpCodedException:
+        """
+        path = self.parent.storage_dir
+        shutil.rmtree(path, ignore_errors=True)
+        try:
+            repository = lib.Repository(path)
+            repository.create()
+        except lib.LibError, le:
+            pe = PulpCodedException(errors.OST0007, path=path, reason=str(le))
```

Comment:
Any chance there could be a non-ascii character in the string representation of the exception?